### PR TITLE
Support controlling levels of movement speed

### PR DESCRIPTION
**Pull request by gabloe**

### CODE REVIEW

File: src/Includes/Camera.hpp [hunk 1]
```diff
@@ -14,13 +16,41 @@ enum MovementDirection {
...
+
+constexpr const char* MovementTypeToString(MovementType t) throw()
+{
+    switch (t)
+    {
+        case MovementType::WALK: return "Walk";
+        case MovementType::JOG: return "Jog";
+        case MovementType::RUN: return "Run";
+        case MovementType::SPRINT: return "Sprint";
+        case MovementType::DEFAULT: return "DEFAULT";
```
Comment (deathly809) [inline src/Includes/Camera.hpp]: Why not just "Default"

File: src/Includes/Camera.hpp [hunk 2]
```diff
@@ -14,13 +16,41 @@ enum MovementDirection {
...
+
 // Default camera values
 const float DEFAULT_YAW         = -90.0f;
 const float DEFAULT_PITCH       =  0.0f;
 const float DEFAULT_SPEED       =  2.5f;
 const float DEFAULT_SENSITIVITY =  0.1f;
 const float DEFAULT_ZOOM        =  45.0f;
 
+// Movement speed ladder threshold
+const float MOVEMENT_SPEED_DELTAT_STEP_THRESHOLD = 0.01f; // Increase the step when time delta since the last incease is above the threshold
```
Comment (deathly809) [inline src/Includes/Camera.hpp]: _DELTAT_?

File: src/Resources/Camera.cpp [hunk 3]
```diff
@@ -94,18 +96,57 @@ void camera::Camera::HandleMouseScroll(float yoffset) {
...
+
+  if (MovementSpeedDeltaT >= MOVEMENT_SPEED_DELTAT_STEP_THRESHOLD) {
+    // Reset the speed time delta and test whether movement speed should increase
+    MovementSpeedDeltaT = 0;
+    MovementSpeedStep += increase; // Update the speed % increase
+    const MovementType movementTypeTest = static_cast<MovementType>(MovementSpeedStep);
+    MovementType newMovementType = MovementType::DEFAULT;
+
+    // Test if the new movement speed is at or above one of the steps
+    for (auto t : AllMovementTypes) {
```
Comment (deathly809) [inline src/Resources/Camera.cpp]: Can't you just do something similar as you did above?

```
  if (MovementSpeedStep >= *std::max_element(AllMovementTypes.begin(), AllMovementTypes.end())) {
```

**Review by deathly809 — CHANGES_REQUESTED** [review]

File: src/Includes/Camera.hpp (hunk 2 above)
Comment (gabloe) [reply src/Includes/Camera.hpp]: Time delta since the last update

Comment (deathly809) [reply src/Includes/Camera.hpp]: Maybe \_DETLA_T_?

**Pull request merged**